types/zpool.nix

```diff
@@ -70,7 +81,8 @@
         in
         {
           dev = ''
-            zpool list '${config.name}' >/dev/null 2>/dev/null || zpool import '${config.name}'
+            zpool list '${config.name}' >/dev/null 2>/dev/null || \
+              zpool import -R ${config.mountRoot} '${config.name}'
```

Comment:
Given that you set the altroot here, during import, is it even necessary to do so above, during creation, as well? The remark about `cachefile=none` still applies here.

Reply:
Yes, because zpool create mounts the pool already.

Reply:
Maybe I am missing something, but why do we need to set altroot during creation? Shouldn't the import suffice to mount it in a live system? I'll try to do a practice test tomorrow

Reply:
It's important to have -R on create too, otherwise any datasets that have `option.mountpoint` set will be mounted when they are created. If altroot is not set, this can shadow existing mountpoints like `/`, which would be fatal.